src/StackExchange.Redis/RedisDatabase.cs

```diff
@@ -3172,32 +3208,99 @@ private Message GetSortedSetCombineAndStoreCommandMessage(SetOperation operation
             {
                 SetOperation.Intersect => RedisCommand.ZINTERSTORE,
                 SetOperation.Union => RedisCommand.ZUNIONSTORE,
+                SetOperation.Difference =>
+                    // Difference can't have weights or aggregation
+                    weights != null || aggregate != Aggregate.Sum
+                    ? throw new ArgumentException("ZDIFFSTORE cannot be used with weights or aggregation.")
+                    : RedisCommand.ZDIFFSTORE,
                 _ => throw new ArgumentOutOfRangeException(nameof(operation)),
             };
-            if (keys == null) throw new ArgumentNullException(nameof(keys));
+            if (keys == null)
+            {
+                throw new ArgumentNullException(nameof(keys));
+            }
+            if (weights != null && keys.Length != weights.Length)
+            {
+                throw new ArgumentException("Keys and weights should have the same number of elements.", nameof(weights));
+            }
 
-            List<RedisValue>? values = null;
-            if (weights != null && weights.Length != 0)
+            RedisValue[] values = RedisValue.EmptyArray;
+
+            var argsLength = (weights?.Length > 0 ? 1 + weights.Length : 0) + (aggregate != Aggregate.Sum ? 2 : 0);
+            if (argsLength > 0)
             {
-                (values ??= new List<RedisValue>()).Add(RedisLiterals.WEIGHTS);
+                values = new RedisValue[argsLength];
+                AddWeightsAggregationAndScore(values, weights, aggregate);
+            }
+            return new SortedSetCombineAndStoreCommandMessage(Database, flags, command, destination, keys, values);
+        }
+
+        private Message GetSortedSetCombineCommandMessage(SetOperation operation, RedisKey[] keys, double[]? weights, Aggregate aggregate, bool withScores, CommandFlags flags)
+        {
+            var command = operation switch
```

Comment:
Probably want to move this logic to the `SetOperation` enum as an extension method.

Reply:
:+1: This makes sense with an arg - done!